shuup/front/utils/cache.py

```diff
@@ -15,32 +15,35 @@
 
 
 def get_listed_products_cache_item(shop):
-    return LISTED_PRODUCTS_CACHE_ITEM_FMT.format(shop_id=shop.pk)
+    shop_id = (shop.id if hasattr(shop, "pk") else shop)
```

Comment:
You can annotate the `shop` parameter to receive either a Shop instance OR an int, this makes it clear to the dev that we are expecting an instance, or the shop id
```py
def get_listed_products_cache_item(shop: Union[Shop, int]):
    shop_id = (shop.pk if isintance(shop, Shop) else shop))
    return LISTED_PRODUCTS_CACHE_ITEM_FMT.format(shop_id=shop_id)
```